src/lib/ngsi/ContextAttributeVector.cpp

```diff
@@ -101,10 +100,23 @@ std::string ContextAttributeVector::toJsonTypes()
 
     std::map<std::string, int>::iterator jt;
     unsigned int                         jx;
+
     for (jt = attrTypes.begin(), jx = 0; jt != attrTypes.end(); ++jt, ++jx)
     {
       std::string type = jt->first;
-      out += JSON_STR(type);
+      
+      //
+      // Special condition for 'options=noAttrDetail':
+      //   When the 'options' URI parameter contains 'noAttrDetail',
+      //   mongoBackend fills the attribute type vector with *just one item* (that is an empty string).
+      //   This special condition is checked for here, to produce a [] for the vector for the response.
+      //
+      // See the origin of this in mongoQueryTypes.cpp. Look for "NOTE: here we add", in two locations.
```

Comment:
Has been this "NOTE: here we add" actually added? I have looked in the PR but I haven't found it...